lib/terrafying/aws.rb

```diff
@@ -307,6 +307,28 @@ def hosted_zone(fqdn)
           end
       end
 
+      def hosted_zone_by_tag(tag)
+        @hosted_zones ||= {}
+        @hosted_zones[tag] ||=
+          begin
+            STDERR.puts "looking for a hosted zone with tag '#{tag}'"
+            hosted_zones = @route53_client.list_hosted_zones().hosted_zones.select { |zone|
+              tags = @route53_client.list_tags_for_resource({resource_type: "hostedzone", resource_id: zone.id.split('/')[2]}).resource_tag_set.tags.select { |aws_tag|
+                aws_tag.key == String(tag.keys[0]) && aws_tag.value == String(tag.values[0])
```

Comment:
it feels a bit weird it accepts a map and then only uses the first value. Can we either explicty pass just one key value or check the whole map?